servicetalk-http-netty/src/testFixtures/java/io/servicetalk/http/netty/ProxyTunnel.java

```diff
@@ -79,12 +86,41 @@ public HostAndPort startProxy() throws IOException {
                 executor.submit(() -> {
                     try {
                         final InputStream in = socket.getInputStream();
-                        final String initialLine = readLine(in);
-                        while (readLine(in).length() > 0) {
-                            // Ignore headers.
+                        final String host;
+                        final int port;
+                        final String protocol;
+                        try {
+                            final String initialLine = readLine(in);
+                            if (!initialLine.startsWith(CONNECT_PREFIX)) {
+                                throw new IllegalArgumentException("Expected " + CONNECT + " request, but found: " +
+                                        initialLine);
+                            }
+                            final int end = initialLine.indexOf(' ', CONNECT_PREFIX.length());
+                            final String authority = initialLine.substring(CONNECT_PREFIX.length(), end);
+                            final int colon = authority.indexOf(':');
+                            host = authority.substring(0, colon);
+                            port = Integer.parseInt(authority.substring(colon + 1));
+                            protocol = initialLine.substring(end + 1);
+
+                            final String hostHeader = readLine(in);
```

Comment:
Where this parsing could potentially be improved is that if I read this correctly it expects the host header to always come as the first header. Others are possible i.e. `Proxy-Authorization` and they might have a different order, but since we use this for internal testing it's fine. Just wanted to mention it in case we want to beef up the testing in the future they could be parsed into a map and then you do your assertions against the map.

Reply:
That's a good observation. You are right, it can have other headers. Will improve later when we test those use-cases.